GPU/GPUTracking/Merger/GPUTPCGMPropagator.cxx

```diff
@@ -155,11 +155,20 @@ GPUd() int GPUTPCGMPropagator::RotateToAlpha(float newAlpha)
   float trackX = x0 * cc + ss * mT->Y();
 
   // transport t0 to trackX
-  float B[3];
-  GetBxByBz(newAlpha, t0.X(), t0.Y(), t0.Z(), B);
   float dLp = 0;
-  if (t0.PropagateToXBxByBz(trackX, B[0], B[1], B[2], dLp)) {
-    return -1;
+  float Bz;
+  if (mPropagateBzOnly) {
+    Bz = GetBz(t0.X(), t0.Y(), t0.Z());
+    if (!t0.PropagateToXBzLight(trackX, Bz, dLp)) {
```

Comment:
@martenole : Here is a typo, the check should not re inverted. Can you please remove the `!` and retry. I did a quick check here and then it looks OK.